src/features/campaigns/components/ActivityList/index.tsx

```diff
@@ -17,12 +17,61 @@ import useClusteredActivities, {
   CLUSTER_TYPE,
 } from 'features/campaigns/hooks/useClusteredActivities';
 import CanvassAssignmentListItem from './items/CanvassAssignmentListItem';
+import ActivityListItem, { STATUS_COLORS } from './items/ActivityListItem';
 
 interface ActivitiesProps {
   activities: CampaignActivity[];
   orgId: number;
 }
 
+interface LazyActivitiesBoxProps extends BoxProps {
+  index: number;
+}
+
+const LazyActivitiesBox = ({
+  index,
+  children,
+  ...props
+}: LazyActivitiesBoxProps) => {
+  const [inView, setInView] = useState(false);
+  const boxRef = useRef<HTMLElement>();
+
+  useEffect(() => {
+    if (!boxRef.current) {
+      return;
+    }
+
+    const observer = new IntersectionObserver((entries) => {
```

Comment:
I'm so glad you introduced this solution. I was not aware of `IntersectionObserver` (like I was with `ResizeObserver` and observers in general) so I had to look it up, but it's covered by all browsers that we care about so I think it's the ultimate solution for us.